src/box/txn_limbo.c

```diff
@@ -575,9 +579,11 @@ txn_limbo_read_demote(struct txn_limbo *limbo, int64_t lsn)
 void
 txn_limbo_ack(struct txn_limbo *limbo, uint32_t replica_id, int64_t lsn)
 {
-	assert(!txn_limbo_is_ro(limbo));
 	if (rlist_empty(&limbo->queue))
 		return;
+	if (limbo->frozen)
+		return;
```

Comment:
It doesn't really look right to completely ignore an ACK while you are waiting. How will it confirm the transaction after unfreeze if you skipped all the acks?

Reply:
AFAIU the idea is to only unfreeze on promote/demote. So we don't really care about losing ACKs.
What do you think?

Reply:
I'd even disallow turning fencing off completely. But Borya told me it's for the sake of some tests.  So I'm ok with it.

Reply:
Yeah, with that way of unfreezing we really don't need acks so far. Thanks for the explanation.